python/kserve/kserve/model_server.py

```diff
@@ -326,8 +326,8 @@ async def stop(self, sig: Optional[int] = None):
         if self._grpc_server:
             logger.info("Stopping the grpc server")
             await self._grpc_server.stop(sig)
-        for _, model in self.registered_models.get_models().items():
-            model.unload()
+        for model_name in list(self.registered_models.get_models().keys()):
```

Comment:
if `get_models()` is returning a map then you only need to do following, as the iterator loop through keys by default
```python
for model_name in self.registered_models.get_models():
      self.registered_models.unload(model_name)
```

Reply:
Python does not allows modifying the dict while iterating, otherwise we will get a runtime error. For example,
Traceback (most recent call last):
  File "main.py", line 11, in <module>
    for model in map:
RuntimeError: dictionary changed size during iteration
`